File: languages/django_models/character.py
from .abstract_element_model import AbstractElementModel
from django.core.validators import MaxValueValidator
from django.db import models

class Character(AbstractElementModel):

    # Constitution
    physicality = models.TextField(blank=True, null=True)
    mentality = models.TextField(blank=True, null=True)
    height = models.PositiveIntegerField(blank=True, null=True)
    weight = models.PositiveIntegerField(blank=True, null=True)
    species = models.ManyToManyField("Species", blank=True, related_name="character_species")
    traits = models.ManyToManyField("Trait", blank=True, related_name="character_traits")
    abilities = models.ManyToManyField("Ability", blank=True, related_name="character_abilities")

    # Origins
    background = models.TextField(blank=True, null=True)
    motivations = models.TextField(blank=True, null=True)
    birth_date = models.PositiveIntegerField(blank=True, null=True)
    birthplace = models.ForeignKey("Location", on_delete=models.SET_NULL, blank=True, null=True, related_name="character_birthplace")
    languages = models.ManyToManyField("Language", blank=True, related_name="character_languages")

    # World
    reputation = models.TextField(blank=True, null=True)
    location = models.ForeignKey("Location", on_delete=models.SET_NULL, blank=True, null=True, related_name="character_location")
    objects = models.ManyToManyField("Object", blank=True, related_name="character_objects")
    institutions = models.ManyToManyField("Institution", blank=True, related_name="character_institutions")

    # Personality
    charisma = models.PositiveIntegerField(validators=[MaxValueValidator(100)], blank=True, null=True)
    coercion = models.PositiveIntegerField(validators=[MaxValueValidator(100)], blank=True, null=True)
    competence = models.PositiveIntegerField(validators=[MaxValueValidator(100)], blank=True, null=True)
    compassion = models.PositiveIntegerField(validators=[MaxValueValidator(100)], blank=True, null=True)
    creativity = models.PositiveIntegerField(validators=[MaxValueValidator(100)], blank=True, null=True)
    courage = models.PositiveIntegerField(validators=[MaxValueValidator(100)], blank=True, null=True)

    # Social
    family = models.ManyToManyField("Family", blank=True, related_name="character_family")
    friends = models.ManyToManyField("Character", blank=True, related_name="character_friends")
    rivals = models.ManyToManyField("Character", blank=True, related_name="character_rivals")

    # Ttrpg
    level = models.PositiveIntegerField(blank=True, null=True)
    STR = models.PositiveIntegerField(blank=True, null=True)
    DEX = models.PositiveIntegerField(blank=True, null=True)
    CON = models.PositiveIntegerField(blank=True, null=True)
    INT = models.PositiveIntegerField(blank=True, null=True)
    WIS = models.PositiveIntegerField(blank=True, null=True)
    CHA = models.PositiveIntegerField(blank=True, null=True)
    hit_points = models.PositiveIntegerField(blank=True, null=True)

    def __str__(self):
        return self.name

File: languages/csharp_unity_json_properties/Character.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using Utils;

[System.Serializable]
public class Character : Element
{
    [JsonProperty("physicality"), TextAttribute("")]
    public string physicality;
    [JsonProperty("mentality"), TextAttribute("")]
    public string mentality;
    [JsonProperty("height"), Integer(0)]
    public int height;
    [JsonProperty("weight"), Integer(0)]
    public int weight;
    [JsonProperty("species"), ReferenceAttribute(typeof(Species), true)]
    public string species;
    [JsonProperty("traits"), ReferenceAttribute(typeof(Trait), true)]
    public string traits;
    [JsonProperty("abilities"), ReferenceAttribute(typeof(Ability), true)]
    public string abilities;
    [JsonProperty("background"), TextAttribute("")]
    public string background;
    [JsonProperty("motivations"), TextAttribute("")]
    public string motivations;
    [JsonProperty("birth_date"), Integer(0)]
    public int birthDate;
    [JsonProperty("birthplace"), ReferenceAttribute(typeof(Location))]
    public string birthplace;
    [JsonProperty("languages"), ReferenceAttribute(typeof(Language), true)]
    public string languages;
    [JsonProperty("reputation"), TextAttribute("")]
    public string reputation;
    [JsonProperty("location"), ReferenceAttribute(typeof(Location))]
    public string location;
    [JsonProperty("objects"), ReferenceAttribute(typeof(Object), true)]
    public string objects;
    [JsonProperty("institutions"), ReferenceAttribute(typeof(Institution), true)]
    public string institutions;
    [JsonProperty("charisma"), Integer(100)]
    public int charisma;
    [JsonProperty("coercion"), Integer(100)]
    public int coercion;
    [JsonProperty("competence"), Integer(100)]
    public int competence;
    [JsonProperty("compassion"), Integer(100)]
    public int compassion;
    [JsonProperty("creativity"), Integer(100)]
    public int creativity;
    [JsonProperty("courage"), Integer(100)]
    public int courage;
    [JsonProperty("family"), ReferenceAttribute(typeof(Family), true)]
    public string family;
    [JsonProperty("friends"), ReferenceAttribute(typeof(Character), true)]
    public string friends;
    [JsonProperty("rivals"), ReferenceAttribute(typeof(Character), true)]
    public string rivals;
    [JsonProperty("level"), Integer(0)]
    public int level;
    [JsonProperty("hit_points"), Integer(0)]
    public int hitPoints;
    [JsonProperty("STR"), Integer(0)]
    public int sTR;
    [JsonProperty("DEX"), Integer(0)]
    public int dEX;
    [JsonProperty("CON"), Integer(0)]
    public int cON;
    [JsonProperty("INT"), Integer(0)]
    public int iNT;
    [JsonProperty("WIS"), Integer(0)]
    public int wIS;
    [JsonProperty("CHA"), Integer(0)]
    public int cHA;
}
